python/sdk/prelude_sdk/controllers/build_controller.py

```diff
@@ -102,6 +105,24 @@ def create_threat(self, name, published, threat_id=None, source_id=None, source=
             return res.json()
         raise Exception(res.text)
 
+    @verify_credentials
+    def create_threat_from_directory(self, directory: str, **kwargs):
+        tests = list()
+        uploads = list()
+        try:
+            for technique_dir in os.listdir(directory):
+                with open(f'{directory}/{technique_dir}/test.go', 'r') as f:
+                    go_code = f.read()
+                with open(f'{directory}/{technique_dir}/config.json', 'r') as f:
+                    config = json.load(f)
+                test = self.create_test(name=config['name'], unit=config['unit'], technique=config['technique'])
+                tests.append(test)
+                uploads.append(self.upload(test_id=test['id'], filename='test.go', data=go_code.encode()))
+        except FileNotFoundError as e:
+            raise Exception(e)
+        threat = self.create_threat(**kwargs, tests=','.join([t['id'] for t in tests]))
+        return {'threat': threat, 'tests': tests, 'uploads': uploads}
```

Comment:
i wonder if we should throw this whole thing in a try/catch so in event of an error half way through, you can still output the uploads that _did_ happen